### PR TITLE
Fix PutTemplateRequest deserialization

### PR DESCRIPTION
**Pull request by Bfindlay**
### Description
Link to issue [417](https://github.com/opensearch-project/opensearch-java/issues/417)

This PR adds in the missing deserialization steps to the PutTemplateRequest _DESERIALIZER which would result in the error
`Missing required property 'PutTemplateRequest.name'` if you tried to deserialize a request object.

### Issues Resolved
https://github.com/opensearch-project/opensearch-java/issues/417

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: java-client/src/test/java/org/opensearch/client/opensearch/core/PutTemplateRequestTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.junit.Test;
+import org.opensearch.client.json.JsonpMapper;
+import org.opensearch.client.json.jsonb.JsonbJsonpMapper;
+import org.opensearch.client.opensearch.indices.PutTemplateRequest;
+
+public class PutTemplateRequestTest extends Assert {
+    private static String INDEX_NAME_FIELD = "name";
+    private static String INDEX_PATTERNS_FIELD = "index_patterns";
+    private static String CREATE_FIELD = "create";
+    private static String ORDER_FIELD = "order";
```
Comment (dblock) [inline java-client/src/test/java/org/opensearch/client/opensearch/core/PutTemplateRequestTest.java:18]: Nit: why do we need these `private static String`? They aren't reused.

File: java-client/src/test/java/org/opensearch/client/opensearch/core/PutTemplateRequestTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Test
+    public void deserialize_validFieldsIncluded_RequestIsBuilt() throws JsonProcessingException {
+        // Arrange
```
Comment (dblock) [inline java-client/src/test/java/org/opensearch/client/opensearch/core/PutTemplateRequestTest.java:22]: IMO this and the other two comments are superfluous and should be removed. Doesn't add any value to the test.

**Review by dblock — COMMENTED** [review]
Thanks! I made some nitpicky comments.

File: java-client/src/test/java/org/opensearch/client/opensearch/core/PutTemplateRequestTest.java (hunk 2 above)
Comment (Bfindlay) [reply java-client/src/test/java/org/opensearch/client/opensearch/core/PutTemplateRequestTest.java:22]: This is just following the usual convention for AAA testing. Happy to remove it if this is not the case here.

File: java-client/src/test/java/org/opensearch/client/opensearch/core/PutTemplateRequestTest.java (hunk 1 above)
Comment (Bfindlay) [reply java-client/src/test/java/org/opensearch/client/opensearch/core/PutTemplateRequestTest.java:18]: I'll update this.

**Comment by Bfindlay** [comment]
@dblock Comments addressed 👍

**Review by reta — APPROVED** [review]

**Review by dblock — APPROVED** [review]

**Comment by Bfindlay** [comment]
Hi @reta @VachaShah I see the backport job failed, does this mean that the fix was not released?

**Comment by reta** [comment]
> Hi @reta @VachaShah I see the backport job failed, does this mean that the fix was not released?

Sadly yeah, we missed it but @VachaShah did manual backport a few days ago. I think we could make another release this week, sorry about that @Bfindlay

**Comment by Bfindlay** [comment]
@reta Thanks for the update, would be great if you can keep me in the loop on the release as this is a blocker for us.
Thanks!

**Pull request merged**